cpp/src/pausable_thread_loop.cpp

```diff
@@ -37,27 +51,40 @@ PausableThreadLoop::~PausableThreadLoop() {
 
 bool PausableThreadLoop::is_running() const noexcept {
     std::lock_guard<std::mutex> lock(mutex_);
-    return !paused_;
+    return state_ != State::Paused;
```

Comment:
```suggestion
    return state_ != State::Paused and state_ != Stopping;
```

Reply:
Stopping is still running (the thread loop might still be executing).